### PR TITLE
Gitignore now working and Fixed a crash issue when user lacked profile picture

### PR DESCRIPTION
**Pull request by eugeneogongo**
Fixes # #1

### CODE REVIEW

**Review by titusgicheru — APPROVED** [review]

File: GraphApp/frmMain.cs [hunk 1]
```diff
@@ -71,14 +73,7 @@ private async void MetroButtonSignIn_Click(object sender, EventArgs e)
...
 
-            if (graphClient != null)
-            {
-                metroButtonSignIn.Text = "Refresh";
-            }
-            else
-            {
-                metroButtonSignIn.Text = "Sign In";
-            }
+            metroButtonSignIn.Text = graphClient != null ? "Refresh" : "Sign In";
```
Comment (titusgicheru) [inline GraphApp/frmMain.cs:76]: Prefer the the blocks, it makes the code readable

File: GraphApp/frmMain.cs [hunk 2]
```diff
@@ -144,11 +138,10 @@ private static async Task<User> GetMeAsync()
 
         private static async Task<Stream> GetMePhotoAsync()
         {
-            Stream profilePhotoStream = null;
             try
             {
                 graphClient = GraphServiceClientProvider.GetAuthenticatedClient();
-                profilePhotoStream = await graphClient.Me.Photo.Content.Request().GetAsync();
+                var profilePhotoStream = await graphClient.Me.Photo.Content.Request().GetAsync();
```
Comment (titusgicheru) [inline GraphApp/frmMain.cs:144]: Use types as opposed to inference, again makes the code readable.

File: GraphApp/frmMain.cs [hunk 3]
```diff
@@ -126,12 +121,11 @@ private static async Task BindProfilePhoto()
 
         private static async Task<User> GetMeAsync()
         {
-            User currentUser = null;
             try
             {
                 graphClient = GraphServiceClientProvider.GetAuthenticatedClient();
                 // Request to get the current logged in user object from Microsoft Graph
-                currentUser = await graphClient.Me.Request().GetAsync();
+                var currentUser = await graphClient.Me.Request().GetAsync();
```
Comment (titusgicheru) [inline GraphApp/frmMain.cs:128]: Same here avoid inference

File: GraphApp/frmMain.cs [hunk 4]
```diff
@@ -172,11 +165,10 @@ private static async Task GetMessages()
 
         private static async Task<IUserMessagesCollectionPage> GetMessagesAsync()
         {
-            IUserMessagesCollectionPage messages = null;
             try
             {
                 graphClient = GraphServiceClientProvider.GetAuthenticatedClient();
-                messages = await graphClient.Me
+                var messages = await graphClient.Me
```
Comment (titusgicheru) [inline GraphApp/frmMain.cs:171]: Same here avoid inference

File: GraphApp/frmMain.cs [hunk 5]
```diff
@@ -210,12 +202,10 @@ private static async Task ListRecentFiles()
 
         private static async Task<IDriveRecentCollectionPage> GetFilesAsync()
         {
-            IDriveRecentCollectionPage driveRecentCollectionPage = null;
-
             try
             {
                 graphClient = GraphServiceClientProvider.GetAuthenticatedClient();
-                driveRecentCollectionPage = await graphClient.Me.Drive.Recent().Request().GetAsync();
+                var driveRecentCollectionPage = await graphClient.Me.Drive.Recent().Request().GetAsync();
```
Comment (titusgicheru) [inline GraphApp/frmMain.cs:208]: Same here avoid inference

File: GraphApp/frmMain.cs (hunk 2 above)
Comment (eugeneogongo) [reply GraphApp/frmMain.cs:144]: Readability is subjective, I actually find using 'var' much more readable than using explicit typing.

Comment (titusgicheru) [reply GraphApp/frmMain.cs:144]: Avoid var when the type is not apparent from the right side of the assignment. You could checkout [C# Coding Conventions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions)

**Pull request closed without merging**